spark/src/main/scala/org/apache/spark/sql/comet/execution/shuffle/CometNativeShuffleWriter.scala

```diff
@@ -140,6 +142,22 @@ class CometNativeShuffleWriter[K, V](
       MapStatus.apply(SparkEnv.get.blockManager.shuffleServerId, partitionLengths, mapId)
   }
 
+  private def isSinglePartitioning(p: Partitioning): Boolean = p match {
```

Comment:
the entire method might be simplified just 
`p.numPartitions <= 1`

?

Reply:
The number of partition bounds can be less than the number of target partitions based on value cardinality, so we still need to check if `rangePartitionBounds.isEmpty`.